datalad_fuse/tests/conftest.py

```diff
@@ -0,0 +1,62 @@
+from pathlib import Path
+import subprocess
+import sys
+import time
+
+from datalad.distribution.dataset import Dataset
+import pytest
+import requests
+
+DATA_DIR = Path(__file__).with_name("data")
+
+LOCAL_SERVER_PORT = 60069
+
+
+@pytest.fixture(scope="session")
```

Comment:
I don't see this anyhow telling that it needs a separate thread or subprocess to run/finish.

Why didn't you just use `@serve_http_path` I have recommended -- doesn't work with pytest?

we have copied (I think without changes IIRC) that code into reproman at some point and even adjusted testing to use pytest and it seems to work:

https://github.com/ReproNim/reproman/blob/HEAD/reproman/tests/test_tests_utils.py#L321

Reply:
@yarikoptic The subprocess spawned by `Popen()` runs in parallel to the main process.  Doing it this way just seemed simpler than `@serve_http_path`.

Reply:
how is it simpler to develop something new instead of using existing/tested?

there is a `while` loop below which runs in the same process as the tests, not even sure how it is supposed to work - works for you locally?

Reply:
re `while` ah, sorry -- it is just to check if server is up. so indeed might work.  I would have probably made that loop to go e.g. for 10 sec and raise an exception if never reaches the "good state".  At least would not stall CI indefinitely (since I guess that is what is happening here)

Reply:
note also that IIRC that `serve_http_path` does ports sweeping as well, so it would find one available (instead of hardcoding to a single one)

Reply:
@yarikoptic It's at least simpler in terms of line count, and I find it easier to comprehend at a glance.

The `while` loop waits for the server to be ready before yielding.  Yes, everything works for me locally.

Reply:
there is no "good" fixed port number to rely on, they need to be sweeped, etc.  Not sure how line count is a winner here if it is just 1 line to use `@serve_http_path`. FWIW -- you can even use it in this fixture in principle if you want a session wide served path/url

Reply:
@yarikoptic I believe I've figured out the problem: The `--directory` option was only added in Python 3.7, yet the workflow is using Python 3.6; hence, the subprocess was failing as soon as it started up.  Regardless, I've switched to (a slightly-stripped down version of) `serve_http_path` for the automatic port allocation benefit.

Reply:
ok, thank you.